File: examples/compose/duis02_nested_view_bad.kt
package examples.compose

import androidx.compose.foundation.layout.*
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

/**
 * DUIS-02 – NESTED VIEW
 *
 * Deeply nested Boxes/Rows/Columns just to show a simple text.
 */
@Composable
fun NestedViewExample() {
    Scaffold { padding ->
        Box(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
        ) {
            Box(modifier = Modifier.padding(8.dp)) {
                Column {
                    Row {
                        Box {
                            Column {
                                Row {
                                    Box {
                                        Column {
                                            Text(
                                                text = "Deeply nested composable tree",
                                                modifier = Modifier.padding(8.dp)
                                            )
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}

File: examples/flutter/duis02_nested_view_bad.dart
import 'package:flutter/material.dart';

/// DUIS-02 – Deeply Nested View Tree (Nested View)
/// - Many nested Containers/Rows/Columns for a simple text.
class NestedLayoutExample extends StatelessWidget {
  const NestedLayoutExample({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        padding: const EdgeInsets.all(8),
        child: Container(
          margin: const EdgeInsets.all(8),
          child: Row(
            children: [
              Expanded(
                child: Container(
                  padding: const EdgeInsets.all(8),
                  child: Column(
                    children: [
                      Container(
                        margin: const EdgeInsets.all(4),
                        child: Row(
                          children: [
                            Expanded(
                              child: Container(
                                padding: const EdgeInsets.all(4),
                                child: Column(
                                  children: [
                                    Container(
                                      padding: const EdgeInsets.all(4),
                                      child: Row(
                                        children: const [
                                          Expanded(
                                            child: Text(
                                              "Deeply nested text",
                                              textAlign: TextAlign.center,
                                            ),
                                          ),
                                        ],
                                      ),
                                    ),
                                  ],
                                ),
                              ),
                            ),
                          ],
                        ),
                      ),
                    ],
                  ),
                ),
              )
            ],
          ),
        ),
      ),
    );
  }
}
